src/ui/components/Difference/index.jsx

```diff
@@ -2,40 +2,130 @@ import React, { useState, useEffect } from "react";
+
+  const { data: diffingResult, isLoading } = getDiffingResultQuery(
+    projectStatus.projectKey,
+    project.beforeVersionId,
+    project.afterVersionId,
+    pageId,
+    projectStatus.accessToken,
+  );
+  const navigate = useNavigate();
   const [displayText, setDisplayText] = useState({
     titleOfChanges: null,
     detailOfChanges: ["변경사항을 선택해주세요."],
     className: "default",
   });
 
   const handleRectangleClick = ev => {
-    setIsLoading(true);
-
     if (ev.data.pluginMessage.type === "RENDER_DIFFERENCE_INFORMATION") {
       const differences = ev.data.pluginMessage.content;
 
-      const differencesInformation = processDifferences(differences);
+      if (differences === "UNCHANGED_NODE") {
+        setDisplayText({
+          titleOfChanges: null,
+          detailOfChanges: ["변경사항을 선택해주세요."],
+          className: "default",
+        });
+      } else {
+        const differencesInformation = processDifferences(differences);
 
-      setDisplayText(differencesInformation);
+        setDisplayText(differencesInformation);
+      }
     }
 
-    setIsLoading(false);
+    if (ev.data.pluginMessage.type === "GET_PROJECT_KEY") {
+      const projectKey = ev.data.pluginMessage.content;
+
+      setProjectStatus(projectStatus => ({ ...projectStatus, projectKey }));
+    }
+
+    if (ev.data.pluginMessage.type === "GET_ACCESS_TOKEN") {
+      const accessToken = ev.data.pluginMessage.content;
+
+      setProjectStatus(projectStatus => ({ ...projectStatus, accessToken }));
+    }
+
+    if (ev.data.pluginMessage.type === "CHANGED_CURRENT_PAGE_ID") {
+      const pageId = ev.data.pluginMessage.content;
+
+      const isComparablePage = allPageIds.includes(pageId);
+      if (!isComparablePage) {
+        setToast({
+          status: true,
+          message: "이전 버전에 존재하지 않는 페이지 입니다.",
+        });
+
+        return;
+      }
+
+      setPageId(pageId);
+    }
   };
 
+  useEffect(() => {
+    setDisplayText({
+      titleOfChanges: null,
+      detailOfChanges: ["변경사항을 선택해주세요."],
+      className: "default",
+    });
+  }, [pageId]);
+
+  useEffect(() => {
+    if (!diffingResult) {
+      return;
+    }
+
+    if (diffingResult.result === "error") {
+      setToast({ status: true, message: diffingResult.message });
+
+      return;
+    }
+
+    const { differences } = diffingResult.content;
+    const modifiedFrames = {};
+
+    for (const frameId in diffingResult.content.frames) {
+      if (isOwnProperty(diffingResult.content.frames, frameId)) {
+        const frameNode = diffingResult.content.frames[frameId];
+
+        modifiedFrames[frameId] = frameNode.property.absoluteBoundingBox;
+        modifiedFrames[frameId].isNew = true;
+      }
+    }
+
+    postMessage("POST_DIFFING_RESULT", { differences, modifiedFrames });
+  }, [diffingResult]);
+
   useEffect(() => {
     window.addEventListener("message", handleRectangleClick);
 
+    postMessage("GET_PROJECT_KEY");
+    postMessage("GET_ACCESS_TOKEN");
+
```

Comment:
P1: 해당 내용 위치를 다른 파일과 동일하게 통일해주시면 좋을 것 같습니다 👍 

```suggestion
  useEffect(() => {
    postMessage("GET_PROJECT_KEY");
    postMessage("GET_ACCESS_TOKEN");
    
    window.addEventListener("message", handleRectangleClick);
```